kura/org.eclipse.kura.ble.provider/src/main/java/org/eclipse/kura/internal/ble/BluetoothLeDeviceImpl.java

```diff
@@ -25,40 +24,41 @@
 import org.eclipse.kura.bluetooth.le.BluetoothLeDevice;
 import org.eclipse.kura.bluetooth.le.BluetoothLeGattService;
 
-import tinyb.BluetoothDevice;
-import tinyb.BluetoothException;
-import tinyb.BluetoothGattService;
+import org.freedesktop.dbus.exceptions.DBusException;
+import org.freedesktop.dbus.types.UInt16;
+
+import com.github.hypfvieh.bluetooth.wrapper.BluetoothDevice;
+import com.github.hypfvieh.bluetooth.wrapper.BluetoothGattService;
 
 public class BluetoothLeDeviceImpl implements BluetoothLeDevice {
 
-    private static final long TIMEOUT = 30;
     private final BluetoothDevice device;
 
-    public BluetoothLeDeviceImpl(tinyb.BluetoothDevice device) {
+    public BluetoothLeDeviceImpl(BluetoothDevice device) {
         this.device = device;
     }
 
     @Override
     public BluetoothLeGattService findService(UUID uuid) throws KuraBluetoothResourceNotFoundException {
-        return findService(uuid, BluetoothLeDeviceImpl.TIMEOUT);
-    }
-
-    @Override
-    public BluetoothLeGattService findService(UUID uuid, long timeout) throws KuraBluetoothResourceNotFoundException {
-        BluetoothGattService service = this.device.find(uuid.toString(), Duration.ofSeconds(timeout));
+        BluetoothGattService service = this.device.getGattServiceByUuid(uuid.toString());
         if (service != null) {
             return new BluetoothLeGattServiceImpl(service);
         } else {
             throw new KuraBluetoothResourceNotFoundException("Gatt service not found.");
         }
     }
 
+    @Override
+    public BluetoothLeGattService findService(UUID uuid, long timeout) throws KuraBluetoothResourceNotFoundException {
+        return this.findService(uuid);
+    }
+
```

Comment:
You removed the timeout functionality from this method. Is this because you are sure that the method will always return or throw an exception? Or it can be possible that the device will get stuck forever?

Reply:
The bluez-dbus library has no such method with a configurable timeout. There is a default timeout associated with the DBus connection (10 seconds) at which point this method will throw an error.